base/reduce.jl

```diff
@@ -1246,3 +1184,84 @@ function _simple_count(::typeof(identity), x::Array{Bool}, init::T=0) where {T}
     end
     return n
 end
+
+# simd optimization for min/max related reduction
+# general fallback
+_fast(op, x, y) = op(x, y)
+_fast(op) = (x, y) -> _fast(op, x, y)
+
+# The following are used in optimized `mapreduce_impl` for IEEEFloat.
+# They compare `0.0/-0.0` correctly.
+_fast(::typeof(min), x::T, y::T) where {T<:IEEEFloat} = ifelse(signbit(x-y), x, y)
+_fast(::typeof(max), x::T, y::T) where {T<:IEEEFloat} = ifelse(signbit(y-x), x, y)
```

Comment:
IIUC, your implementation of `_fast` have different pre- and post-conditions than the pre-PR version of `_fast`. In particular, this now eliminates the need for the "bad zero" fix-up (narrower post-condition) while it now requires the explicit nan check (stricter pre-condition).

Can you add a docstring to `_fast`, explaining this? It'd help other contributors understand the new usage pattern of `_fast`. It's better to put this in the docstring than comment since it's more discoverable in the REPL.

Reply:
Docstring has been added.